omnibus/files/private-chef-ctl-commands/reindex.rb

```diff
@@ -102,7 +102,16 @@ def do_reindex(orgs_to_reindex, options)
 
   OptionParser.new do |opts|
     opts.on("-w", "--wait", "Wait for reindex queue to clear before exiting") do |w|
-      options[:wait] = w
+      # Elasticsearch doesn't do any queue waiting, so -w doesn't do anything
+      if running_config["private_chef"]["opscode-erchef"]["search_provider"] == "elasticsearch"
+        $stderr.puts <<-EOF
+Elasticsearch is currently configured as the search provider.
+It does not support waiting for reindex queue. Ignoring "wait" option.
```

Comment:
I think this is more whether search mode is `rabbitmq` vs `batch`/`inline`.  The `rabbitmq` mode is the only time this flag is applicable.
ref: https://github.com/chef/chef-server/blob/master/omnibus/files/private-chef-cookbooks/private-chef/attributes/default.rb#L320